client/state/jetpack-settings/modules/actions.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * External dependencies
+ */
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	JETPACK_MODULE_ACTIVATE,
+	JETPACK_MODULE_ACTIVATE_FAIL,
+	JETPACK_MODULE_ACTIVATE_SUCCESS
+} from 'state/action-types';
+import wp from 'lib/wp';
+
+export const activateModule = ( module_slug ) => {
+	return ( dispatch ) => {
+		dispatch( {
+			type: JETPACK_MODULE_ACTIVATE,
+			module: module_slug
+		} );
+		return new Promise( ( resolve, reject ) => {
```

Comment:
Ha, I didn't know that method exists already. Good find! Let's use it here, it appears to be exactly what we need.